sources/Debate/src/main/java/com/example/Debate/controller/SearchController.java

```diff
@@ -0,0 +1,36 @@
+package com.example.Debate.controller;
+
+import com.example.Debate.dto.response.FullDebateResponseDto;
+import com.example.Debate.service.SearchService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin
+@RequestMapping("/search")
+public class SearchController {
+
+    private SearchService searchService;
+
+    public SearchController(SearchService searchService){this.searchService = searchService;}
+
+    @GetMapping("/byName")
+    public ResponseEntity<List<FullDebateResponseDto>> getDebatesContainingName(@RequestParam String reqName)
+    {
+        List<FullDebateResponseDto> debateDtoList = searchService.getDebatesDebatesContainingName(reqName);
+        return debateDtoList.size() != 0 ? ResponseEntity.status(HttpStatus.OK).body(debateDtoList) :
+                ResponseEntity.status(HttpStatus.NOT_FOUND).body(debateDtoList);

```

Comment:
In REST APIs when we are returning an empty collection, most of the times we don't return HTTP 404 Not Found error. 404 means that the server could not locate the specified resource. When asking for collections, or searching for them, we do not ask for a specific resource, but a collection - it could be an empty collection, that is completely fine. Also, the client can parse the response easier - returning 404 would be seen as an error and would have to be correctly handled, while returning an empty array usually complies with the client's code, or at the very least is much easier to handle than 404 error.
See https://softwareengineering.stackexchange.com/a/358293 for further explanation

Reply:
So, we do not need to check if our List<FullDebateResponseDto> is empty - we can just return it :)